ntbs-service/Models/FormattedDate.cs

```diff
@@ -20,6 +20,17 @@ public bool HasAnyEmpty()
             return string.IsNullOrEmpty(Year) || string.IsNullOrEmpty(Month) || string.IsNullOrEmpty(Day);
         }
 
+        public static FormattedDate Today()

```

Comment:
I'm not convinced that we need this. Seems like the default date should be applied at creation time, then there's no need for custom handling of the null type in what is effectively just a view model. My feeling is that we should be setting the underlying DateTime field to today's date

Reply:
Notification could have been created a couple days ago and left there and then completed later on. I don't think setting Notification Date on Notification Creating will work. But if you mean setting it when initialising the constructor, then we face an issue line 53 on Episode.cshtml.cs, as it will get notificationDate as null from database

Reply:
Do you mean here?
https://github.com/PublicHealthEngland/ntbs_Beta/blob/13c57973c8e32a859b12689dc461250bbd5cf3ee/ntbs-service/Pages/Notifications/Edit/Episode.cshtml.cs#L53
I don't think that's an issue. I'm suggesting that we store the NotificationDate in the database as soon as the notification draft is created, effectively stopping it form ever being null.

I agree that there's a scenario where someone creates the draft on one day and submits it 2 days later, but I think it's more logical to do it like this. In current implementation, we are allowing them to not insert the date, but we are making the experience very awkward:
1. I create a draft on on Monday, go to hospital details page and see the date filled in to today's date
1. Most of the time, I would leave it be (because it's already there and I have no reason to change it. It gets saved to the database with that Moday's date.
1. The end result is *effectively* the same as if we used what I'm suggesting and just saved that date to the db at creation time.

Second possibility is if I *DO* decide to wait on that date and not save the default:
1. I create a draft on on Monday, go to hospital details page and see the date filled in to today's date
1. I *clear* the date fields
1. I save the page
1. I come back to that page and it shows me today's date - it *looks like* the date has actually been persisted and clearing it has failed!! I wouldn't notice that *not* being the case until I visited this page on Tuesday